cfgmgr/buffermgrdyn.cpp

```diff
@@ -1013,12 +1052,19 @@ task_process_status BufferMgrDynamic::doRemovePgTask(const string &pg_key, const
     // recalculate pool size
     checkSharedBufferPoolSize();
 
-    if (!portInfo.speed.empty() && !portInfo.cable_length.empty())
-        portInfo.state = PORT_READY;
-    else
-        portInfo.state = PORT_INITIALIZING;
-    SWSS_LOG_NOTICE("try removing the original profile %s", bufferPg.running_profile_name.c_str());
-    releaseProfile(bufferPg.running_profile_name);
+    if (portInfo.state != PORT_DISABLED)
+    {
+        if (!portInfo.speed.empty() && !portInfo.cable_length.empty())
+            portInfo.state = PORT_READY;
+        else
+            portInfo.state = PORT_INITIALIZING;
+    }
+
+    if (!bufferPg.running_profile_name.empty())
```

Comment:
Add comment: In case the port has been admin down, the running_profile_name can be empty.